File: lib/pb/data/share.defs.dart
import 'package:protod/delta.dart' as defs;
import 'package:groupshare/pb/data/foo/foo.defs.dart' as data_foo_foo;

Share_type Share() {
  return Share_type([]);
}

class Share_type extends defs.Locator {
  Share_type(List<defs.Indexer> location) : super(location);

  defs.String_scalar Id() {
    return defs.String_scalar([...location]..add(defs.Field("id", 1)));
  }

  defs.String_scalar Name() {
    return defs.String_scalar([...location]..add(defs.Field("name", 2)));
  }

  data_foo_foo.Foo_type Foo() {
    return data_foo_foo.Foo_type([...location]..add(defs.Field("foo", 3)));
  }

}

class Share_type_repeated extends defs.Locator {
  Share_type_repeated(List<defs.Indexer> location) : super(location);
  Share_type Index(int i) {
    return Share_type([...location]..add(defs.Index(i)));
  }
}

File: server/pb/data/data_defs/share.defs.go
package data_defs

import (
	"github.com/dave/groupshare/server/pb/data/foo/foo_defs"
	"github.com/dave/protod/delta"
)

func Share() Share_type {
	return Share_type{}
}

type Share_type struct {
	location []delta.Indexer
}

func NewShare_type(l []delta.Indexer) Share_type {
	return Share_type{location: l}
}
func (b Share_type) Location_get() []delta.Indexer {
	return b.location
}

type Share_type_repeated struct {
	location []delta.Indexer
}

func NewShare_type_repeated(l []delta.Indexer) Share_type_repeated {
	return Share_type_repeated{location: l}
}
func (b Share_type_repeated) Location_get() []delta.Indexer {
	return b.location
}
func (b Share_type_repeated) Index(i int) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, delta.IndexIndexer(i)))
}
func (b Share_type) Id() delta.String_scalar {
	return delta.NewString_scalar(delta.CopyAndAppend(b.location, delta.FieldIndexer("id", 1)))
}
func (b Share_type) Name() delta.String_scalar {
	return delta.NewString_scalar(delta.CopyAndAppend(b.location, delta.FieldIndexer("name", 2)))
}
func (b Share_type) Foo() foo_defs.Foo_type {
	return foo_defs.NewFoo_type(delta.CopyAndAppend(b.location, delta.FieldIndexer("foo", 3)))
}
